web_src/js/features/repo-diff.ts

```diff
@@ -234,17 +232,80 @@ function initRepoDiffShowMore() {
   });
 }
 
+function shouldStopLoading() {
+  const hash = window.location.hash;
+
+  if (!hash) {
+    return true; // No hash to look for
+  }
+
+  // eslint-disable-next-line unicorn/prefer-query-selector
+  const targetElement = document.getElementById(hash.substring(1));
```

Comment:
Still prefer `querySelector` but not `getElementById`

eg: `document.querySelector(hash)` ?

Reply:
I know, but in this case, the result can be controlled by the user (look below, there's a hashchange listener).
If we don't use getElementById explicitly, false positives/negatives could happen.

Reply:
Why? What's the case to cause "false positives/negatives could happen"?

Reply:
The user manually changes the page hash to, I don't know `#file-input%20.ui.button` or something like that.

Reply:
Which, if my guess is correct, can lead to the false element being selected

Reply:
But you can also input something like `#navbar` `#mobile-notifications-icon` `#navbar-expand-toggle` `#issue-title-display` and a lot of others to make `getElementById` select other "false elements" too, right?

What's the difference?

Reply:
True.
The difference would be that at least this weird case would then be prevented.
However, I do agree that users doing something like that probably deserve whatever they get.

Reply:
In that case, `querySelector` should be fine.

Reply:
Hmm, I think I found an edge case which makes `getElementById` win ..... `document.querySelector('#')` throws JS exception 🤣  If we don't want to use `try-catch` .... then let's keep the original `getElementById` and add some comments.

Thank you thoughtful suggestion  👍